Resources/Catalog/FileCatalog.py

```diff
@@ -209,10 +255,10 @@ def w_execute( self, *parms, **kws ):
           failed.setdefault( lfn, {} )[catalogName] = errorMessage
       # Restore original lfns if they were changed by normalization
       if lfnMapDict:
-        for lfn in failed:
-          failed[lfnMapDict.get( lfn, lfn )] = failed[lfn]
-        for lfn in successful:
-          successful[lfnMapDict.get( lfn, lfn )] = successful[lfn]
+        for lfn in failed.keys():
```

Comment:
Yes because you delete elements of the dictionnary on which you loop 

In [1]: d = dict.fromkeys(range(5))

In [2]: d
Out[2]: {0: None, 1: None, 2: None, 3: None, 4: None}

In [3]: for k in d:
   ...:     d.pop(k)
   ...:     
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-3-ce2d839d70b1> in <module>()
----> 1 for k in d:
      2     d.pop(k)
      3 

RuntimeError: dictionary changed size during iteration